plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -108,19 +108,6 @@ bool exception_is_exhausted(const fc::exception& e) {
 }
 } // namespace
 
-struct transaction_id_with_expiry {
-   transaction_id_type trx_id;
-   fc::time_point      expiry;
-};
-
-struct by_id;
-struct by_expiry;
-
-using transaction_id_with_expiry_index = multi_index_container<
-   transaction_id_with_expiry,
-   indexed_by<hashed_unique<tag<by_id>, BOOST_MULTI_INDEX_MEMBER(transaction_id_with_expiry, transaction_id_type, trx_id)>,
-              ordered_non_unique<tag<by_expiry>, BOOST_MULTI_INDEX_MEMBER(transaction_id_with_expiry, fc::time_point, expiry)>>>;
-
```

Comment:
`producer_plugin` no longer needs to include boost multi index or have the multi index aliases.